source/robot_model/include/robot_model/Model.hpp

```diff
@@ -282,11 +284,17 @@ class Model {
    */
   void set_robot_name(const std::string& robot_name);
 
+  /**
+   * @brief Getter of the URDF string
+   * @return the URDF
+   */
+  const std::string& get_urdf() const;
+
   /**
    * @brief Getter of the URDF path
    * @return the URDF path
    */
-  const std::string& get_urdf_path() const;
+  std::optional<std::reference_wrapper<const std::string>> get_urdf_path() const;
```

Comment:
I went for this because I think it's nicer when you want to indirectly check if the urdf was initialized from a file, but we could simply return an empty string and agree this means there was not file. 

The optional would return `None` for python bindings which might or not be ok for us.